packages/sources/finage-test/src/endpoint/forex.ts

```diff
@@ -4,6 +4,40 @@ import overrides from '../config/overrides.json'
 import { httpTransport } from '../transport/forex-http'
 import { ForexBaseEndpointTypes, forexPriceInputParameters } from './utils'
 import { wsTransport } from '../transport/forex-ws'
+import { AdapterRequest } from '@chainlink/external-adapter-framework/util'
+import { PriceAdapterRequest } from '../index'
+
+/* 
+to improve data quality for forex feeds
+
+On receiving FOO/USD request, if FOO is on the list of assets that are queried in the standard direction (EUR, GBP, AUD, NZD, XAG, XAU, XPT, XPD, XCU)
+ then handle as normal, else invert the request
+
+excludesMap includes currency pairs {quote: [base ...]} which should not be inverted.
+*/
+const excludesMap: Record<string, string[]> = {
+  USD: ['EUR', 'GBP', 'AUD', 'NZD', 'XAG', 'XAU', 'XPT', 'XPD', 'XCU'],
+}
+
+export const forexReqTransformer = (req: AdapterRequest<any>): void => {
```

Comment:
Setting this type will avoid needing to cast base and quote to `String` later

```suggestion
export const forexReqTransformer = (req: AdapterRequest<typeof forexPriceInputParameters.validated>): void => {
```